CRM/Core/Payment/PayPalIPN.php

```diff
@@ -353,6 +353,8 @@ public function main() {
         return;
       }
 
+      $input['payment_processor_id'] = $paymentProcessorID;
+
       self::$_paymentProcessor = &$objects['paymentProcessor'];
```

Comment:
This line can go too & we can stop passing $paymentProcessorID into validateData a few lines earlier

Reply:
@eileenmcnaughton I'm less familiar with the IPN code here - do you want to do a separate PR for just those changes you are suggesting and I'll review?

Reply:
@mattwire ok

Reply:
so far these are the prs to do the follow up on the ipn classes that we left out of scope here 

https://github.com/civicrm/civicrm-core/pull/18540
https://github.com/civicrm/civicrm-core/pull/18600

I don't just want to dive in & pull out the var until the params are a bit more sanitised